conf/config.go

```diff
@@ -127,6 +127,80 @@ type DualRootfsDeviceConfig struct {
 	RootfsPartB string
 }
 
+// Client configuration
+
+// HttpsClient holds the configuration for the client side mTLS configuration
+// NOTE: Careful when changing this, the struct is exposed directly in the
+// 'mender.conf' file.
+type HttpsClient struct {
+	Certificate string `json:",omitempty"`
+	Key         string `json:",omitempty"`
+	SSLEngine   string `json:",omitempty"`
+}
+
+// Security structure holds the configuration for the client
+// Added for MEN-3924 in order to provide a way to specify PKI params
+// outside HttpsClient.
+// NOTE: Careful when changing this, the struct is exposed directly in the
+// 'mender.conf' file.
+type Security struct {
+	AuthPrivateKey string `json:",omitempty"`
+	SSLEngine      string `json:",omitempty"`
+}
+
+// Connectivity instructs the client how we want to treat the keep alive connections
+// and when a connection is considered idle and therefore closed
+// NOTE: Careful when changing this, the struct is exposed directly in the
+// 'mender.conf' file.
+type Connectivity struct {
+	// If set to true, there will be no persistent connections, and every
+	// HTTP transaction will try to establish a new connection
+	DisableKeepAlive bool `json:",omitempty"`
+	// A number of seconds after which a connection is considered idle and closed.
+	// The longer this is the longer connections are up after the first call over HTTP
+	IdleConnTimeoutSeconds int `json:",omitempty"`
+}
+
+func (h *HttpsClient) Validate() {
+	if h == nil {
+		return
+	}
+	if h.Certificate != "" || h.Key != "" {
+		if h.Certificate == "" {
+			log.Error(
+				"The 'Key' field is set in the mTLS configuration, but no 'Certificate' is given." +
+					" Both need to be present in order for mTLS to function",
+			)
+		}
+		if h.Key == "" {
+			log.Error(
+				"The 'Certificate' field is set in the mTLS configuration, but no 'Key' is given." +
+					" Both need to be present in order for mTLS to function",
+			)
+		} else if strings.HasPrefix(h.Key, Pkcs11URIPrefix) && len(h.SSLEngine) == 0 {
+			log.Errorf("The 'Key' field is set to be loaded from %s, but no 'SSLEngine' is given."+
+				" Both need to be present in order for loading of the key to function",
+				Pkcs11URIPrefix)
+		}
+	}
+}
+
+// MenderServer is a placeholder for a full server definition used when
+// multiple servers are given. The fields corresponds to the definitions
+// given in MenderConfig.
+type MenderServer struct {
+	ServerURL string
+	// TODO: Move all possible server specific configurations in
+	//       MenderConfig over to this struct. (e.g. TenantToken?)
+}
+
+type Config struct {
```

Comment:
In this package now, `conf.Config` doesn't really describe what this struct holds. When it was `client.Config` made more sense.

If it is not too much work, can we rename to something explicit like `conf.ClientConfig` or `conf.HttpConfig`?

Reply:
Went for `HttpConfig`